### ISSUE TITLE
PyChess automatically start to observe a FICS game started with same gamenumber what was may last played too

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#831]
Original [issue 831](https://code.google.com/p/pychess/issues/detail?id=831) reported by [gbtami](https://code.google.com/u/gbtami/) 2014-01-26

**What steps will reproduce the problem?**
1. Played a FICS game (game number 9)
2. finished it
3. did nothing, but PyChess started to observe a new game played by others (game number 9)

{Game 9 (gbtami vs. Hamdimoussax) gbtami forfeits on time} 0-1

Lightning rating adjustment: 1325 --> 1316
Lightning rank:  1864/2054  (down 7)
Lightning hrank:  1828/2016  (down 7)
13:58:11 ('gbtami', 'raw') INFO: 94 observe 9
13:58:11 ('gbtami', 'raw') INFO: 95 moves 9
13:58:11 Default DEBUG: ICPlayer.__boardUpdate: id(self)=176674772 self=Hamdimoussax (1396) 9 gbtami Hamdimoussax 62 0 Be4 8/pp3pkp/6p1/8/2P1b1P1/1P1R3P/PK1B1r2/8 w - - 1 32 -111 6413
13:58:11 Default DEBUG: ICPlayer.__boardUpdate: id=176674772 self=Hamdimoussax (1396) gameno=9: this is my move
13:58:11 ('gbtami', 'onStyle12') DEBUG: <12> -------- pp---pkp ------p- -------- --P-b-P- -P-R---P PK-B-r-- -------- W -1 0 0 0 0 1 9 gbtami Hamdimoussax 1 1 0 13 13 -111 6413 32 B/c6-e4 (0:01.469) Be4 0 1 302
13:58:11 ('gbtami', 'nonmatched') DEBUG: {Game 9 (gbtami vs. Hamdimoussax) gbtami forfeits on time} 0-1
13:58:11 ('gbtami', 'nonmatched') DEBUG: Lightning rating adjustment: 1325 --> 1316
13:58:11 ('gbtami', 'nonmatched') DEBUG: Lightning rank:  1864/2054  (down 7)
13:58:11 ('gbtami', 'nonmatched') DEBUG: Lightning hrank:  1828/2016  (down 7)
13:58:11 ('Ping', '13:01:35.837') DEBUG: 64 bytes from fics.freechess.org (69.36.243.188): icmp_seq=216 ttl=52 time=287 ms
13:58:12 ('gbtami', 'raw') DEBUG: 
GuestWTTQ(U)(53):  i suppose you could do the botox
fics% 
<s> 52 w=Suryansah ti=00 rt=1906  t=20 i=5 r=r tp=standard c=B rr=0-9999 a=f f=f
fics% 
<sr> 42
fics% 
<sr> 145
fics% 9037Finger of gbtami:

On for: 3 mins   Idle: 0 secs

```
      rating     RD      win    loss    draw   total   best
```

Blitz      1557     40.7     523     574      70    1167   1569 (26-Nov-2013)
Standard   1840    171.1      11      33       2      46
Lightning  1316     64.0      18      45       1      64   1375 (11-Jan-2014)
Wild       1750    121.6       7      10       0      17
Crazyhouse 1395    107.6       5      17       0      22
Suicide    1567    242.1       0       2       0       2
Losers     1417    272.1       0       1       0       1
Atomic     1476    245.6       0       2       0       2

Email      : btami@enternet.hu

Total time online: 16 days, 9 hrs, 44 mins
% of life online:  0.7  (since Mon Mar 26, 07:53 PDT 2007)

Timeseal 1 : On


fics% 
<s> 102 w=Kingofthekings ti=00 rt=1613  t=3 i=0 r=r tp=blitz c=? rr=1500-2800 a=t f=f
fics% 9156seekinfo set.
<sc>
<s> 3 w=jianjiankangkang ti=00 rt=1739  t=30 i=30 r=r tp=standard c=W rr=1700-9999 a=f f=f
<s> 32 w=domvar ti=00 rt=1898  t=10 i=12 r=r tp=standard c=? rr=1786-9999 a=t f=f
<s> 37 w=donyamihi ti=00 rt=1331  t=3 i=0 r=r tp=blitz c=? rr=1100-1650 a=f f=t
<s> 52 w=Suryansah ti=00 rt=1906  t=20 i=5 r=r tp=standard c=B rr=0-9999 a=f f=f

13:58:12 ('gbtami', 'lines') DEBUG: 
13:58:12 ('gbtami', 'lines') DEBUG: GuestWTTQ(U)(53):  i suppose you could do the botox
13:58:12 ('gbtami', 'onChannelMessage') DEBUG: GuestWTTQ(U)(53):  i suppose you could do the botox
13:58:12 ('gbtami', 'lines') DEBUG: 
13:58:12 ('gbtami', 'lines') DEBUG: <s> 52 w=Suryansah ti=00 rt=1906  t=20 i=5 r=r tp=standard c=B rr=0-9999 a=f f=f
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 52 w=Suryansah ti=00 rt=1906  t=20 i=5 r=r tp=standard c=B rr=0-9999 a=f f=f
13:58:12 ('gbtami', 'lines') DEBUG: 
13:58:12 ('gbtami', 'lines') DEBUG: <sr> 42
13:58:12 ('gbtami', 'on_seek_remove') DEBUG: <sr> 42
13:58:12 ('gbtami', 'lines') DEBUG: 
13:58:12 ('gbtami', 'lines') DEBUG: <sr> 145
13:58:12 ('gbtami', 'on_seek_remove') DEBUG: <sr> 145
13:58:12 ('gbtami', 'command_reply') DEBUG: 90 37 Finger of gbtami:

On for: 3 mins   Idle: 0 secs

rating     RD      win    loss    draw   total   best
Blitz      1557     40.7     523     574      70    1167   1569 (26-Nov-2013)
Standard   1840    171.1      11      33       2      46
Lightning  1316     64.0      18      45       1      64   1375 (11-Jan-2014)
Wild       1750    121.6       7      10       0      17
Crazyhouse 1395    107.6       5      17       0      22
Suicide    1567    242.1       0       2       0       2
Losers     1417    272.1       0       1       0       1
Atomic     1476    245.6       0       2       0       2

Email      : btami@enternet.hu

Total time online: 16 days, 9 hrs, 44 mins
% of life online:  0.7  (since Mon Mar 26, 07:53 PDT 2007)

Timeseal 1 : On
13:58:12 ('gbtami', 'lines') DEBUG: Finger of gbtami:

On for: 3 mins   Idle: 0 secs

rating     RD      win    loss    draw   total   best
Blitz      1557     40.7     523     574      70    1167   1569 (26-Nov-2013)
Standard   1840    171.1      11      33       2      46
Lightning  1316     64.0      18      45       1      64   1375 (11-Jan-2014)
Wild       1750    121.6       7      10       0      17
Crazyhouse 1395    107.6       5      17       0      22
Suicide    1567    242.1       0       2       0       2
Losers     1417    272.1       0       1       0       1
Atomic     1476    245.6       0       2       0       2

Email      : btami@enternet.hu

Total time online: 16 days, 9 hrs, 44 mins
% of life online:  0.7  (since Mon Mar 26, 07:53 PDT 2007)

Timeseal 1 : On
13:58:12 ('gbtami', 'lines') DEBUG: 
13:58:12 ('gbtami', 'lines') DEBUG: <s> 102 w=Kingofthekings ti=00 rt=1613  t=3 i=0 r=r tp=blitz c=? rr=1500-2800 a=t f=f
13:58:12 Default DEBUG: Ignored fingerline: 'rating     RD      win    loss    draw   total   best'
13:58:12 ('Ping', '13:01:12.283') DEBUG: /bin/ping
13:58:12 (u'Stockfish', '13:01:11.130') DEBUG: info depth 15 seldepth 31 score cp 48 nodes 1000571 nps 993615 time 1007 multipv 1 pv f2h2 b2c1 h2h3 d2g5 f7f5 d6d7 g7f8 g4f5 g6f5 g5f4 f8e8 d7g7 h7h5 f4e5 a7a6 c1d2 h3d3 d2e2 d3d7 g7d7 e8d7 e2e3
13:58:12 ('gbtami', 'onFinger') DEBUG: Finger of gbtami:

On for: 3 mins   Idle: 0 secs

rating     RD      win    loss    draw   total   best
Blitz      1557     40.7     523     574      70    1167   1569 (26-Nov-2013)
Standard   1840    171.1      11      33       2      46
Lightning  1316     64.0      18      45       1      64   1375 (11-Jan-2014)
Wild       1750    121.6       7      10       0      17
Crazyhouse 1395    107.6       5      17       0      22
Suicide    1567    242.1       0       2       0       2
Losers     1417    272.1       0       1       0       1
Atomic     1476    245.6       0       2       0       2

Email      : btami@enternet.hu

Total time online: 16 days, 9 hrs, 44 mins
% of life online:  0.7  (since Mon Mar 26, 07:53 PDT 2007)

Timeseal 1 : On

13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 102 w=Kingofthekings ti=00 rt=1613  t=3 i=0 r=r tp=blitz c=? rr=1500-2800 a=t f=f
13:58:12 ('gbtami', 'raw') DEBUG: <s> 98 w=JackRed ti=01 rt=0P t=10 i=0 r=u tp=blitz c=? rr=0-9999 a=t f=f
<s> 102 w=Kingofthekings ti=00 rt=1613  t=3 i=0 r=r tp=blitz c=? rr=1500-2800 a=t f=f
<s> 126 w=FSCTLohi ti=00 rt=1871  t=35 i=0 r=r tp=standard c=B rr=0-1850 a=f f=t

fics% 9210You are not playing a game.

fics% 93103You are not playing a game.

fics% 
13:58:12 ('gbtami', 'command_reply') DEBUG: 91 56 seekinfo set.
<sc>
<s> 3 w=jianjiankangkang ti=00 rt=1739  t=30 i=30 r=r tp=standard c=W rr=1700-9999 a=f f=f
<s> 32 w=domvar ti=00 rt=1898  t=10 i=12 r=r tp=standard c=? rr=1786-9999 a=t f=f
<s> 37 w=donyamihi ti=00 rt=1331  t=3 i=0 r=r tp=blitz c=? rr=1100-1650 a=f f=t
<s> 52 w=Suryansah ti=00 rt=1906  t=20 i=5 r=r tp=standard c=B rr=0-9999 a=f f=f
<s> 98 w=JackRed ti=01 rt=0P t=10 i=0 r=u tp=blitz c=? rr=0-9999 a=t f=f
<s> 102 w=Kingofthekings ti=00 rt=1613  t=3 i=0 r=r tp=blitz c=? rr=1500-2800 a=t f=f
<s> 126 w=FSCTLohi ti=00 rt=1871  t=35 i=0 r=r tp=standard c=B rr=0-1850 a=f f=t
13:58:12 ('gbtami', 'lines') DEBUG: seekinfo set.
<sc>
<s> 3 w=jianjiankangkang ti=00 rt=1739  t=30 i=30 r=r tp=standard c=W rr=1700-9999 a=f f=f
<s> 32 w=domvar ti=00 rt=1898  t=10 i=12 r=r tp=standard c=? rr=1786-9999 a=t f=f
<s> 37 w=donyamihi ti=00 rt=1331  t=3 i=0 r=r tp=blitz c=? rr=1100-1650 a=f f=t
<s> 52 w=Suryansah ti=00 rt=1906  t=20 i=5 r=r tp=standard c=B rr=0-9999 a=f f=f
<s> 98 w=JackRed ti=01 rt=0P t=10 i=0 r=u tp=blitz c=? rr=0-9999 a=t f=f
<s> 102 w=Kingofthekings ti=00 rt=1613  t=3 i=0 r=r tp=blitz c=? rr=1500-2800 a=t f=f
<s> 126 w=FSCTLohi ti=00 rt=1871  t=35 i=0 r=r tp=standard c=B rr=0-1850 a=f f=t
13:58:12 ('gbtami', 'nonmatched') DEBUG: seekinfo set.
13:58:12 ('gbtami', 'on_seek_clear') DEBUG: <sc>
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 3 w=jianjiankangkang ti=00 rt=1739  t=30 i=30 r=r tp=standard c=W rr=1700-9999 a=f f=f
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 32 w=domvar ti=00 rt=1898  t=10 i=12 r=r tp=standard c=? rr=1786-9999 a=t f=f
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 37 w=donyamihi ti=00 rt=1331  t=3 i=0 r=r tp=blitz c=? rr=1100-1650 a=f f=t
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 52 w=Suryansah ti=00 rt=1906  t=20 i=5 r=r tp=standard c=B rr=0-9999 a=f f=f
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 98 w=JackRed ti=01 rt=0P t=10 i=0 r=u tp=blitz c=? rr=0-9999 a=t f=f
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 102 w=Kingofthekings ti=00 rt=1613  t=3 i=0 r=r tp=blitz c=? rr=1500-2800 a=t f=f
13:58:12 ('gbtami', 'on_seek_add') DEBUG: <s> 126 w=FSCTLohi ti=00 rt=1871  t=35 i=0 r=r tp=standard c=B rr=0-1850 a=f f=t
13:58:12 ('gbtami', 'command_reply') DEBUG: 92 10 You are not playing a game.
13:58:12 ('gbtami', 'lines') DEBUG: You are not playing a game.
13:58:12 ('gbtami', 'nonmatched') DEBUG: You are not playing a game.
13:58:12 ('gbtami', 'command_reply') DEBUG: 93 103 You are not playing a game.
13:58:12 ('gbtami', 'lines') DEBUG: You are not playing a game.
13:58:12 ('gbtami', 'nonmatched') DEBUG: You are not playing a game.
13:58:12 ('GuestVTRL', 'raw') DEBUG: 
{Game 418 (Gargornon vs. Cirov) Cirov forfeits by disconnection} 1-0
fics% 
Gargornon Blitz (1894), Std (1940), Wild (1792), Light(1564), Bug(----)
  is now available for matches.
fics% 
<wd> Cirov
fics% 
<wa> Cirov 001717 1887E0P0P0P0P0P0P0P
fics% 
Cirov Blitz (1717), Std (1887), Wild (----), Light(----), Bug(----)
  is now available for matches.
fics% 
<wa> mised 010P0P0P0P0P0P0P0P0P
fics% 
13:58:12 ('GuestVTRL', 'lines') DEBUG: 
13:58:12 ('GuestVTRL', 'lines') DEBUG: {Game 418 (Gargornon vs. Cirov) Cirov forfeits by disconnection} 1-0
13:58:12 Default DEBUG: BM.onGameEnd: <FICSGame wplayer=<FICSPlayer name='Gargornon', online=True, adjournment=False, status=10, game.gameno=418, game.rated=True, game.private=False, Schnell=1887, Standard=1940, VillÃ¡m=1564, Atom=1604, Tandem=0, VisszarakÃ³s=1692, LÃºzer=0, Ãngyilkos=0, Vad=1792>, bplayer=<FICSPlayer name='Cirov', online=True, adjournment=False, status=10, game.gameno=418, game.rated=True, game.private=False, Schnell=1722, Standard=1887, VillÃ¡m=0, Atom=0, Tandem=0, VisszarakÃ³s=0, LÃºzer=0, Ãngyilkos=0, Vad=0>, gameno=418, game_type=<GameType fics\_name='blitz', short\_fics\_name='b', rating\_type=0, variant\_type=0, display\_text='Schnell'>, rated=True, min=3, inc=2, result=4, reason=26, private=False>
13:58:12 ('GuestVTRL', 'on_game_remove') DEBUG: {Game 418 (Gargornon vs. Cirov) Cirov forfeits by disconnection} 1-0
13:58:12 ('GuestVTRL', 'lines') DEBUG: 
13:58:12 ('GuestVTRL', 'lines') DEBUG: Gargornon Blitz (1894), Std (1940), Wild (1792), Light(1564), Bug(----)
13:58:12 ('GuestVTRL', 'lines') DEBUG: is now available for matches.
13:58:12 ('GuestVTRL', 'on_player_available') DEBUG: Gargornon Blitz (1894), Std (1940), Wild (1792), Light(1564), Bug(----)
is now available for matches.
13:58:12 ('GuestVTRL', 'lines') DEBUG: 
13:58:12 ('GuestVTRL', 'lines') DEBUG: <wd> Cirov
13:58:12 ('GuestVTRL', 'on_player_disconnect') DEBUG: <wd> Cirov
13:58:12 ('GuestVTRL', 'lines') DEBUG: 
13:58:12 ('GuestVTRL', 'lines') DEBUG: <wa> Cirov 001717 1887E0P0P0P0P0P0P0P
13:58:12 ('GuestVTRL', 'on_player_connect') DEBUG: <wa> Cirov 001717 1887E0P0P0P0P0P0P0P
13:58:12 ('GuestVTRL', 'lines') DEBUG: 
13:58:12 ('GuestVTRL', 'lines') DEBUG: Cirov Blitz (1717), Std (1887), Wild (----), Light(----), Bug(----)
13:58:12 ('GuestVTRL', 'lines') DEBUG: is now available for matches.
13:58:12 ('GuestVTRL', 'on_player_available') DEBUG: Cirov Blitz (1717), Std (1887), Wild (----), Light(----), Bug(----)
is now available for matches.
13:58:12 ('GuestVTRL', 'lines') DEBUG: 
13:58:12 ('GuestVTRL', 'lines') DEBUG: <wa> mised 010P0P0P0P0P0P0P0P0P
13:58:12 ('GuestVTRL', 'on_player_connect') DEBUG: <wa> mised 010P0P0P0P0P0P0P0P0P
13:58:12 ('gbtami', 'raw') DEBUG: 9480You are now observing game 9.
Game 9: JohnHolland (1253) falak (1402) rated blitz 3 0

<12> rnbqkbnr pppppppp -------- -------- -------- -------- PPPPPPPP RNBQKBNR W -1 1 1 1 1 0 9 JohnHolland falak 0 3 0 39 39 180000 180000 1 none (0:00.000) none 0 0 0

fics% 9577
Movelist for game 9:

JohnHolland (1253) vs. falak (1402) --- Sun Jan 26, 04:58 PST 2014
Rated blitz match, initial time: 3 minutes, increment: 0 seconds.

Move  JohnHolland             falak              

---

```
  {Still in progress} *
```


fics% 
13:58:12 ('gbtami', 'command_reply') DEBUG: 94 80 You are now observing game 9.
Game 9: JohnHolland (1253) falak (1402) rated blitz 3 0

<12> rnbqkbnr pppppppp -------- -------- -------- -------- PPPPPPPP RNBQKBNR W -1 1 1 1 1 0 9 JohnHolland falak 0 3 0 39 39 180000 180000 1 none (0:00.000) none 0 0 0
13:58:12 ('gbtami', 'lines') DEBUG: You are now observing game 9.
Game 9: JohnHolland (1253) falak (1402) rated blitz 3 0

<12> rnbqkbnr pppppppp -------- -------- -------- -------- PPPPPPPP RNBQKBNR W -1 1 1 1 1 0 9 JohnHolland falak 0 3 0 39 39 180000 180000 1 none (0:00.000) none 0 0 0
13:58:12 ('gbtami', 'nonmatched') DEBUG: You are now observing game 9.
13:58:12 ('gbtami', 'nonmatched') DEBUG: Game 9: JohnHolland (1253) falak (1402) rated blitz 3 0
13:58:12 ('gbtami', 'onStyle12') DEBUG: <12> rnbqkbnr pppppppp -------- -------- -------- -------- PPPPPPPP RNBQKBNR W -1 1 1 1 1 0 9 JohnHolland falak 0 3 0 39 39 180000 180000 1 none (0:00.000) none 0 0 0
13:58:12 ('gbtami', 'command_reply') DEBUG: 95 77 Movelist for game 9:

JohnHolland (1253) vs. falak (1402) --- Sun Jan 26, 04:58 PST 2014
Rated blitz match, initial time: 3 minutes, increment: 0 seconds.

Move  JohnHolland             falak

---

{Still in progress} *
13:58:12 ('gbtami', 'lines') DEBUG: Movelist for game 9:

JohnHolland (1253) vs. falak (1402) --- Sun Jan 26, 04:58 PST 2014
Rated blitz match, initial time: 3 minutes, increment: 0 seconds.

Move  JohnHolland             falak

---

{Still in progress} *
13:58:12 Default DEBUG: ICLounge.onObserveGameCreated: <FICSGame wplayer=<FICSPlayer name='JohnHolland', online=True, adjournment=False, status=0, game.gameno=9, game.rated=True, game.private=False, Schnell=1253, Standard=1468, VillÃ¡m=1041, Atom=0, Tandem=0, VisszarakÃ³s=1515, LÃºzer=1599, Ãngyilkos=0, Vad=0>, bplayer=<FICSPlayer name='falak', online=True, adjournment=False, status=0, game.gameno=9, game.rated=True, game.private=False, Schnell=1402, Standard=1479, VillÃ¡m=1504, Atom=0, Tandem=0, VisszarakÃ³s=1475, LÃºzer=0, Ãngyilkos=0, Vad=0>, gameno=9, game_type=<GameType fics\_name='blitz', short\_fics\_name='b', rating\_type=0, variant\_type=0, display\_text='Schnell'>, rated=True, min=3, inc=0, private=False>
13:58:12 Default DEBUG: ionest.generalStart: <ICGameModel at 2853518708 (ply=0, variant=Standard, status=0, reason=35, ficsgame=<FICSGame wplayer=<FICSPlayer name='JohnHolland', online=True, adjournment=False, status=0, game.gameno=9, game.rated=True, game.private=False, Schnell=1253, Standard=1468, VillÃ¡m=1041, Atom=0, Tandem=0, VisszarakÃ³s=1515, LÃºzer=1599, Ãngyilkos=0, Vad=0>, bplayer=<FICSPlayer name='falak', online=True, adjournment=False, status=0, game.gameno=9, game.rated=True, game.private=False, Schnell=1402, Standard=1479, VillÃ¡m=1504, Atom=0, Tandem=0, VisszarakÃ³s=1475, LÃºzer=0, Ãngyilkos=0, Vad=0>, gameno=9, game_type=<GameType fics\_name='blitz', short\_fics\_name='b', rating\_type=0, variant\_type=0, display\_text='Schnell'>, rated=True, min=3, inc=0, private=False>, players=[], tags={'TimeControl': '180+0', 'Time': '13:58:00', 'Result': '*', 'Round': 1, 'Year': 2014, 'Month': 1, 'Day': 26, 'Event': 'FICS rated blitz game', 'Site': 'FICS'}

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=831#c1) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2014-01-29

Could you post the whole log or at least, more of it?

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=831#c2) posted by [gbtami](https://code.google.com/u/gbtami/) 2014-01-29

Unfortunately no, I not saved it :(

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=831#c3) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2014-01-29

I wonder if we should increase the default number of log files saved. 10 seems low. I've lost log files with relevant debugging info in them many times because they got auto-deleted.

**Comment by pychess** [comment]
Original [comment 4.](https://code.google.com/p/pychess/issues/detail?id=831#c4) posted by [gbtami](https://code.google.com/u/gbtami/) 2014-01-29

Agreed.

**Comment by pychess** [comment]
Original [comment 5.](https://code.google.com/p/pychess/issues/detail?id=831#c5) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2014-02-09
- changed status to: Started

**Comment by pychess** [comment]
Original [comment 6.](https://code.google.com/p/pychess/issues/detail?id=831#c6) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2014-02-09

Fixed in revision d7f13af261c5
- changed status to: Fixed
